File: backend/entity/Advisor.go
package entity

import "gorm.io/gorm"

type SelectAdvisor struct {
	gorm.Model
	No uint `json:"no" valid:"required,range(1|20)~Number must be between 1 and 20"`
	// Title string `json:"title" valid:"required~Title is required"`
	Description string `json:"description" valid:"required~Description is required"`
	Status      string `json:"status" valid:"required~Status is required" gorm:"default:'pending'"`

	GroupProjectID uint          `json:"group_project_id" valid:"required~GroupProjectID is required"`
	GroupProject   *GroupProject `gorm:"foreignKey:GroupProjectID" json:"group_project"`
	TeacherID      uint          `json:"teacher_id" valid:"required~TeacherID is required"`
	Teacher        *User         `gorm:"foreignKey:TeacherID" json:"teacher"`
}

File: frontend/app/interfaces/Advisor.ts
import { User,GroupProject } from "./Group";

export interface Teacher extends User {
}

export interface AdvisorSelectionPayload {
    group_project_id: number;
    description: string;
    advisor_order: number[]; 
}

export interface SelectAdvisor {
    ID: number;
    CreatedAt?: string;
    UpdatedAt?: string;
    DeletedAt?: string | null;

    no: number;             
    description: string;    
    status: string;        

    group_project_id: number;
    group_project?: GroupProject; 

    teacher_id: number;
    teacher?: Teacher;     
}


export interface AdvisorActionRequest {
  selection_id: number;
}

export interface AdvisorStatusRequest {
  is_open: boolean;
}